node/state.re

```diff
@@ -129,26 +129,16 @@ let try_to_commit_state_hash = (~old_state, state, block, signatures) => {
 
 let apply_block = (state, block) => {
   let old_state = state;
-  let.ok (protocol, new_snapshot, results) =
-    apply_block(state.protocol, block);
+  let.ok (protocol, results) = apply_block(state.protocol, block);
```

Comment:
We should probably check here to see if block.state_root_hash matches the exped block.state_root_hash

Reply:
We already introduced a check in #227 for this: https://github.com/marigold-dev/deku/pull/227/files#diff-43dae84d72a098cf13a754e41ae4fc754d43ee859e8be911f36ae612b542aaa5R194-R199

Or do you mean something else?